base/compiler/stmtinfo.jl

```diff
@@ -47,24 +47,26 @@ function nmatches(info::UnionSplitInfo)
     return n
 end
 
-struct ConstResult
+struct ConcreteResult
     mi::MethodInstance
     effects::Effects
     result
-    ConstResult(mi::MethodInstance, effects::Effects) = new(mi, effects)
-    ConstResult(mi::MethodInstance, effects::Effects, @nospecialize val) = new(mi, effects, val)
+    ConcreteResult(mi::MethodInstance, effects::Effects) = new(mi, effects)
+    ConcreteResult(mi::MethodInstance, effects::Effects, @nospecialize val) = new(mi, effects, val)
 end
 
+const ConstResult = Union{InferenceResult,ConcreteResult}
```

Comment:
This doesn't really make sense to me ... an `InferenceResult` is not always a `Const`.